mountpoint-s3-client/tests/copy_object.rs

```diff
@@ -0,0 +1,77 @@
+#![cfg(feature = "s3_tests")]
+
+pub mod common;
+use aws_sdk_s3::primitives::ByteStream;
+use bytes::Bytes;
+use common::creds::get_sdk_default_chain_creds;
+use common::*;
+use mountpoint_s3_client::config::{EndpointConfig, S3ClientAuthConfig, S3ClientConfig};
+use mountpoint_s3_client::error::ObjectClientError;
+use mountpoint_s3_client::S3RequestError;
+use mountpoint_s3_client::{ObjectClient, S3CrtClient};
+use mountpoint_s3_crt::auth::credentials::{CredentialsProvider, CredentialsProviderStaticOptions};
+use mountpoint_s3_crt::common::allocator::Allocator;
+#[tokio::test]
+async fn test_copy_objects() {
+    let sdk_client = get_test_sdk_client().await;
+    let (bucket, prefix) = get_test_bucket_and_prefix("test_copy_object_prefix1");
+
+    let key = format!("{prefix}/hello");
+    let body = b"hello world!";
+    sdk_client
+        .put_object()
+        .bucket(&bucket)
+        .key(&key)
+        .body(ByteStream::from(Bytes::from_static(body)))
+        .send()
+        .await
+        .unwrap();
+
+    let credentials = get_sdk_default_chain_creds().await;
+
+    // Build a S3CrtClient that uses a static credentials provider with the creds we just got
+    let config = CredentialsProviderStaticOptions {
+        access_key_id: credentials.access_key_id(),
+        secret_access_key: credentials.secret_access_key(),
+        session_token: credentials.session_token(),
+    };
+    let provider = CredentialsProvider::new_static(&Allocator::default(), config).unwrap();
+    let config = S3ClientConfig::new()
+        .auth_config(S3ClientAuthConfig::Provider(provider))
+        .endpoint_config(EndpointConfig::new(&get_test_region()));
+    let client = S3CrtClient::new(config).unwrap();
+
+    let copy_prefix = get_unique_test_prefix("test_copy_object_prefix2");
+    let copy_key = format!("{copy_prefix}/hello2");
```

Comment:
No need to get another unique prefix, just use a different key (or nested prefixes if you want)
```suggestion
    let copy_key = format!("{prefix}/hello2");
```

Reply:
ack